File: server-java-issuetracker-restapi/src/main/java/hu/elte/issuetrackerrest/entities/Issue.java
package hu.elte.issuetrackerrest.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import java.time.LocalDateTime;
import java.util.List;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class Issue {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private String title;

    @Column
    private String description;

    @Column
    private String place;

    public enum Status {
        NEW, DOING, DONE
    }

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Status status;

    @Column(updatable = false)
    @CreationTimestamp
    private LocalDateTime created_at;

    @Column
    @UpdateTimestamp
    private LocalDateTime updated_at;
    
    @OneToMany(mappedBy = "issue")
    private List<Message> messages;
    
    @ManyToMany
    @JoinTable
    private List<Label> labels;
    
    @ManyToOne
    @JsonIgnore
    private User user;
}

File: server-typescript-issuetracker-restapi/src/entities/issue.ts
import { Entity, PrimaryKey, Property, Enum, Collection, ManyToMany, ManyToOne, OneToMany } from '@mikro-orm/core';
import { Label } from './label';
import { Message } from './message';
import { User } from './user';

@Entity()
export class Issue {

  @PrimaryKey()
  id!: number;

  @Property()
  description!: string;

  @Property()
  title!: string;

  @Property()
  place!: string;

  @Enum()
  status: IssueStatus = IssueStatus.NEW;

  @Property()
  createdAt: Date = new Date();

  @Property({ onUpdate: () => new Date() })
  modifiedAt: Date = new Date();

  @ManyToMany(() => Label, 'issues', { owner: true })
  labels = new Collection<Label>(this);

  @ManyToOne(() => User)
  user!: User;

  @OneToMany(() => Message, message => message.issue)
  messages = new Collection<Message>(this);
}

export enum IssueStatus {
  NEW = 'NEW',
  DOING = 'DOING',
  DONE = 'DONE',
}
